add ai concepts section (#8087)
aws-amplify/docs: src/pages/[platform]/ai/concepts/architecture/index.mdx

import { getCustomStaticPath } from "@/utils/getCustomStaticPath";

export const meta = {
  title: "Architecture",
  description:
    "Amplify AI Kit fullstack architecture",
  platforms: [
    "javascript",
    "react-native",
    "angular",
    "nextjs",
    "react",
    "vue",
  ],
};

export const getStaticPaths = async () => {
  return getCustomStaticPath(meta.platforms);
};

export function getStaticProps(context) {
  return {
    props: {
      platform: context.params.platform,
      meta,
    },
  };
}

The Amplify AI kit is built around the idea of routes. An AI route is like an API endpoint for interacting with backend AI functionality. AI routes are configured in an Amplify backend where you can define the authorization rules, what type of route (generation or conversation), AI model and inference configuration like temperature, what are the inputs and outputs, and what data it has access to. There are currently 2 types of AI routes:

* **Conversation:** A conversation route is an asynchronous, multi-turn API. Conversations and messages are automatically stored in DynamoDB. Examples of this are any chat-based AI experience or conversational UI.

> * **Generation:** A single synchronous request-response API. A generation route is just an AppSync Query. Examples of this are: generating alt text for an image, generating structured data from unstructured input, summarization, etc.

Reviewer: ```suggestion
* **Generation:** A single synchronous request-response API. A generation route is an AppSync Query that generates structured data according to your route definition. Common uses include generating structured data from unstructured input and summarization.
```

## Cloud infrastructure

When you create an AI route with the Amplify AI kit, it is using these services:

### AWS AppSync

> Acts as an API layer to authorize and route requests from the browser to AWS services. 

Reviewer: ```suggestion
Serverless API layer to authorize and route requests from the browser to AWS services. 
```
Just a thought, feel free to ignore. 
It sticks out that the other component descriptions state "Serverless" and AppSync doesn't.

### Amazon DynamoDB
Serverless database for storing conversation history.

### AWS Lambda
Serverless execution for conversations. 

### Amazon Bedrock

> Serverless foundation models

Reviewer: ```suggestion
Serverless foundation models.
```
